pkg/hubble/metrics/flow/handler.go

```
@@ -92,3 +111,7 @@ func (h *flowHandler) ProcessFlow(ctx context.Context, flow *flowpb.Flow) error
 	h.flows.WithLabelValues(labels...).Inc()
 	return nil
 }
+
+func (h *flowHandler) Deinit(registry *prometheus.Registry) {
+	registry.Unregister(h.flows)
```

I figured I would just confirm what Prometheus does when we unregister, and it might not work exactly as expected: https://github.com/prometheus/client_golang/issues/203. We should make sure to test the case where a handler is registered, de-registered, then registered again, and verify what metrics are exported if we scrape the metrics server between each register/deregister/register step.

---

We should also check the error from `Unregister` and if it doesn't deregister we may want to return an error on `Deinit` since it would mean re-registering in the future could potentially panic.

---

Added the following passing test: re-register a deregistered handler with the same config; after dereg the corresponding metrics are gone, and after re-reg they're collected again on ProcessFlow.

Looking at `Unregister` it only returns `false` if it's called on a registry that doesn't contain a record of registration of the handler being deregistered, in which case we should still remove the handler from our list regardless of the error. Still, I changed the `Deinit` signature to return bool.